assets/js/datasets.js

```diff
@@ -0,0 +1,248 @@
+        pub.type = 'inproceedings';
+    } else if (pub.type == 'tutorial') {
+        pub.type = 'misc';
+
+        // Put a note in the bibtex for tutorials.
+        if (!pub.note) {
+            pub.note = '';
+        }
+        pub.note = (pub.note.trim() + ' Tutorial').trim();
+    }
+
+    // Cleanup the year.
+
+    if (!pub.year) {
+        pub.year = 0;
+    }
+    pub.year = parseInt(pub.year, 10);
+
+    // Cleanup and index the authors.
+
+    pub['_authorIDs'] = [];
+    pub.authors.forEach(function(authorName) {
+        authorName = window.linqs.pubs.cleanText(authorName);
+
+        let nameParts = authorName.split(' ');
+        let authorId = window.linqs.pubs.cleanId(authorName);
+
+        pub['_authorIDs'].push(authorId);
+
+        if (authorId in authors) {
+            return;
+        }
+
+        authors[authorId] = {
+            'displayName': authorName,
+            'firstName': nameParts[0],
+            'lastName': nameParts[nameParts.length - 1],
+            'sortName': `${nameParts[nameParts.length - 1]}-${nameParts[0]}-${authorId}`.toLowerCase(),
+        }
+    });
+
+    pub['_sortAuthor'] = authors[pub['_authorIDs'][0]]['sortName']
+
+    window.linqs.pubs.authors = authors;
+}
+
+window.linqs.datasets.bibtex = function(pub, id) {
+    window.linqs.datasets.index(pub, id);
+    return window.linqs.pubs.bibtex(pub);
+};
+
+window.linqs.datasets.size = function(size) {
+    let sizes = ["B", "K", "M", "G", "T"];
+
+    let i = 0;
+    while (i < sizes.length && size >= 1000) {
+        size = size / 1000.0;
+        i++;
+    }
+
+    return size.toFixed(1).toString().replace('.0', '') + sizes[i];
+}
+
+window.linqs.datasets.makeDownloadInfo = function(downloads) {
+    let downloadInfosHTML = '';
+
+    downloads.forEach(function(download, i) {
+        let link = download['download-link'];
+        let text = download["text"];
+        let statsHTML = "";
+
+        let sizeRaw = download['size'];
+        let md5 = download['md5'];
+
+        if (sizeRaw != null && md5 != null) {
+            let size = window.linqs.datasets.size(sizeRaw);
+
+            statsHTML = `
+                <div class='size'>
+                    (${size})
+                </div>
+                <div class='hash'>
+                    md5: ${md5}
+                </div>
+            `;
+        }
+
+        let downloadInfoHTML = `
+            <div class='download-info'>
+                <a class='download-link' href='${link}'>${text}</a>
+                ${statsHTML}
+            </div>
+        `;
+
+        downloadInfosHTML += downloadInfoHTML
+    });
+
+    return downloadInfosHTML;
+}
+
+window.linqs.datasets.makeReferences = function(references) {
+    let referencesHTML = '';
+
+    references.forEach(function(reference, i) {
+        let refLink = reference['link'];
+        let refText = reference['text'];
+        let referenceTemplate = `
+            <p><a href='${refLink}'>${refText}</a></p>
```

Comment:
Why `p`?

Reply:
I added this p to separate out the individual links.

Reply:
Remember what we discussed about <p> tags bvefore.
They are used to represent paragraphs, which are supposed to be on their own line.
Why would you need to put each link in their own paragraph, they are already in their own <a> tag.

Also, don't resolve threads that you comment on (unless you are 100% sure it is resolved).